### ISSUE TITLE
Possible timing issue between jx promote and jx-resource jenkins plugin

### ISSUE THREAD

**Issue by pmuir** [jenkins-x/jx#1890]
### Summary 

Running a PR build for the jx project I got

```
 Created Pull Request: https://github.beescloud.com/****/environment-pr-1888-2-staging/pull/1
        
        Failed to update PipelineActivity: Operation cannot be fulfilled on pipelineactivities.jenkins.io "jenkins-x-tests-bdd-test-spring-1538663830-master-1": the object has been modified; please apply your changes to the latest version and try again
        error: Operation cannot be fulfilled on pipelineactivities.jenkins.io "jenkins-x-tests-bdd-test-spring-1538663830-master-1": the object has been modified; please apply your changes to the latest version and try again
```

### Steps to reproduce the behavior

Unknown

**Comment by msvticket** [comment]
I often get the same error master builds as well. An example:
```
+ jx promote -b --all-auto --timeout 1h --version 1.0.10
Using helmBinary helm with feature flag: template-mode
Promoting app userprofile version 1.0.10 to namespace core-staging
Created Pull Request: https://github.com/deniedboarding/environment-jx-staging/pull/88

Failed to update PipelineActivity: Operation cannot be fulfilled on pipelineactivities.jenkins.io "deniedboarding-userprofile-master-30": the object has been modified; please apply your changes to the latest version and try again
error: Operation cannot be fulfilled on pipelineactivities.jenkins.io "deniedboarding-userprofile-master-30": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by wbrefvem** [comment]
+1

For me it's happening pretty consistently when trying to auto-promote a `jx create spring` application.

**Comment by msvticket** [comment]
The pull request is working for me. It doesn't guarantee this doesn't happen, but it should lessen the likelihood.

I tested by doing `make linux` in the jx repo and then make a docker image using this Dockerfile:
```
FROM jenkinsxio/builder-maven:0.1.21

COPY ./build/linux/jx /usr/bin/jx
```
I pushed the image to a local repo and reconfigered the Jenkins-X Jenkins to use this image for maven builds.

**Comment by msvticket** [comment]
Today it happened to work mate during manual promote as well.

**Comment by msvticket** [comment]
This seems to be fixed by with #3467
